### ISSUE TITLE
Regression with writing to swift container between version 1.18.0 and 1.19.0-alpha3

### ISSUE THREAD

**Issue by vasuarv** [kubernetes/kops#9933]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Version 1.19.0-alpha.3 (git-04b9f41daa)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
v1.19.1

**3. What cloud provider are you using?**
openstack

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
export KOPS_STATE_STORE=swift://ois

kops create cluster --cloud openstack \
        --name my-cluster.k8s.local   \
        --state ${KOPS_STATE_STORE} \
        --zones nova \
        --network-cidr 192.168.0.0/24 \
        --image Ubuntu-20.04-focal \
        --master-count=1 \
        --node-count=1 \
        --node-size 8124120 \
        --master-size 8124120 \
        --topology private \
        --bastion \
        --networking calico \
        --api-loadbalancer-type public \
        --ssh-public-key ~/sshkey.pub \
        --os-ext-net ext-net-vlan52 \
        --os-kubelet-ignore-az=true 

**5. What happened after the commands executed?**
The creation of the cluster failed. The container was created on the openstack cluster. However, the container is empty.

**6. What did you expect to happen?**
The k8s cluster state gets created on the openstack container.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
```
I0913 15:58:54.225023    1573 context.go:283] hit maximum retries 5 with error error writing swift://ois/my-cluster.k8s.local/config: Resource not found

error writing updated configuration: error writing Cluster "my-cluster.k8s.local": error writing configuration file swift://ois/my-cluster.k8s.local/config: error writing swift://ois/my-cluster.k8s.local/config: Resource not found
```
**9. Anything else do we need to know?**
I am in the process of testing out the fix addressed by https://github.com/kubernetes/kops/issues/9893 locally. However, this issue is blocking my testing and validation of that issue. 
Creating the cluster state was not an issue with the current official release Version 1.18.0 (git-698bf974d8). I am seeing it with the current top of the tree Version 1.19.0-alpha.3 (git-04b9f41daa).

**Comment by olemarkus** [comment]
The logs here are not much for me to go on. As far as I know, we have not done anything in this area (other than updating dependencies).

**Comment by kesslerdev** [comment]
Same read/write regression with 1.19.0-beta.2:

```bash
$ kops version
Version 1.19.0-beta.2 (git-c006d97e5596024f9b9a5681c97300165156319e)
$ kops get --name kops.k8s.local -o yaml -v=10
I1211 13:55:44.813617   23127 factory.go:68] state store swift://kops-state-store
I1211 13:55:44.813753   23127 swiftfs.go:66] authenticating to keystone
I1211 13:55:48.084967   23127 swiftfs.go:427] Reading file "swift://kops-state-store/kops.k8s.local/config"

Cluster.kops.k8s.io "kops.k8s.local" not found
```
with 1.18.2:
```bash
$ kops version
Version 1.18.2 (git-84495481e4)
$ kops get --name kops.k8s.local -o yaml -v=10
I1211 13:59:08.283331   23206 factory.go:68] state store swift://kops-state-store
I1211 13:59:08.283788   23206 swiftfs.go:68] authenticating to keystone
I1211 13:59:08.536208   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/config"
I1211 13:59:12.743176   23206 swiftfs.go:467] Listed files in swift://kops-state-store/kops.k8s.local/instancegroup: [swift://kops-state-store/kops.k8s.local/instancegroup/bastions swift://kops-state-store/kops.k8s.local/instancegroup/master-***-1 swift://kops-state-store/kops.k8s.local/instancegroup/master-***-2 swift://kops-state-store/kops.k8s.local/instancegroup/master-***-3 swift://kops-state-store/kops.k8s.local/instancegroup/nodes]
I1211 13:59:12.743310   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/instancegroup/bastions"
I1211 13:59:13.116749   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/instancegroup/master-***-1"
I1211 13:59:13.204781   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/instancegroup/master-***-2"
I1211 13:59:13.335658   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/instancegroup/master-***-3"
I1211 13:59:13.410385   23206 swiftfs.go:424] Reading file "swift://kops-state-store/kops.k8s.local/instancegroup/nodes"
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: "2020-12-11T13:18:08Z"
  name: kops.k8s.local
spec:
  api:
    loadBalancer:
      type: Public
  authorization:
    rbac: {}
  channel: stable
  cloudConfig:
    openstack:
...
```

**Comment by cardoe** [comment]
This appears to be still broken with 1.20. It might be related to this commit in gophercloud... https://github.com/Lirt/gophercloud/commit/e54703519a9c3bcf1e3c25450a02614beb583c6d

**Comment by cardoe** [comment]
@olemarkus Can we reopen this issue and update the version of gophercloud once the fix lands upstream?

**Comment by cardoe** [comment]
Need to update gophercloud to at least https://github.com/gophercloud/gophercloud/commit/b7d5b2cdd7ffc13e79d924f61571b0e5f74ec91c to resolve this issue.

**Comment by cardoe** [comment]
@olemarkus Could I get an "ok-to-test" on the PR to fix this? It's #11788. I'm just bumping the version of gophercloud to the next release which includes a fix for this issue.
